### PR TITLE
Adding error reporting model and basic functions

### PR DESCRIPTION
**Pull request by seancavanaugh**
Just a heads up - We're currently not populating these, but it's available for future use now.

### CODE REVIEW

**Comment by daotoo** [comment]
Do we want an api method for specific dock_ids as well? To be able to view the history of errors from a specific dock over a period of time?

**Comment by seancavanaugh** [comment]
@daotoo Should be added now

**Comment by daotoo** [comment]
The `README.md` should be updated as well with these new end points.

**Comment by seancavanaugh** [comment]
@daotoo how about now?

**Comment by daotoo** [comment]
Yeah I agree, just thought we should add it for now until we overhaul it.

**Comment by daotoo** [comment]
+1

@brnsmns13

File: app/routes/router.js [hunk 1]
```diff
@@ -92,6 +92,16 @@ router
     .get('/findalltransactionsbyparamid', function (req, res){
         API.findTransactionsByParamId(req, res)
     })
+	
+	// gets all the errors
+    .get('/findallerrorreports', function (req, res){
```
Comment (brnsmns13) [inline app/routes/router.js]: Can we change this endpoint to just ```/errors```? These long endpoint names are overly verbose and the action is defined by the HTTP method.

File: app/routes/router.js [hunk 2]
```diff
@@ -92,6 +92,16 @@ router
...
+        API.findAllErrorReports(req, res)
+    })
+	
+    .get('/finderrorreportsbydockid/:dockID', function (req, res){
```
Comment (brnsmns13) [inline app/routes/router.js]: Same as above, change to ```/errors/:dockID```.

File: app/routes/router.js (hunk 1 above)
Comment (seancavanaugh) [inline app/routes/router.js]: @brnsmns13  This seems like a code clarity issue, if It really bothers you we can change it, but at this point I'm more worried about functionality. Ideally, you'd be using the README.MD anyways - though we do need to update this to clarify some information.

Comment (brnsmns13) [inline app/routes/router.js]: @seancavanaugh Yes it is just a code quality issue, but I think we should be shooting for high quality code. Following standard REST practices, the endpoint url should represent the object being interacted with, and the query would specify whether we are getting all data or filtering. I know it really isn't that big of a deal, but small things like this still shouldn't be overlooked.

Comment (seancavanaugh) [inline app/routes/router.js]: @brnsmns13 We'll fix all of the endpoints in a different PR.

Comment (brnsmns13) [inline app/routes/router.js]: @seancavanaugh ok, this is fine for now as long as we get the endpoints and docs updated soon.

**Comment by jcurtiss** [comment]
:+1:

**Pull request merged**